src/datachain/lib/dc.py

```diff
@@ -430,7 +430,11 @@ def parse_uri(
         if listings:
             if update:
                 # choosing the smallest possible one to minimize update time
-                listing = sorted(listings, key=lambda ls: len(ls.name))[0]
+                listing = sorted(listings, key=lambda ls: len(ls.name), reverse=True)[0]
```

Comment:
@ilongin why `parse_uri` is exposed externally? also it's a bit of a strange name tbh. Could you please give more context?

Reply:
Yes, this should not be exposed. Probably this can be removed from `DataChain` completely in the first place ...

Reply:
Kk, I'll move it